third_party/move/move-vm/types/src/values/values_impl.rs

```diff
@@ -1831,10 +1942,39 @@ impl Value {
         ))))
     }
 
-    // REVIEW: This API can break
-    pub fn vector_for_testing_only(it: impl IntoIterator<Item = Value>) -> Self {
-        Self(ValueImpl::Container(Container::Vec(Rc::new(RefCell::new(
-            it.into_iter().map(|v| v.0).collect(),
+    /// Creates a vector of values. Use with caution. It is the caller's responsibility to ensure
+    /// that the values have same types.
```

Comment:
*the* same types

The logic of _unchecked makes sense, but also we added an additional check for IndexRef here :) I am thinking if there is some good naming we can use, since the unchecked part is cross-element invariants, while there always can be checks on individual elements (in fact, it should probably be assumed that all checks will be done).

Reply:
fixed the typo

for the naming, I don't have any ideas, any suggestions? In the meantime I imporved the comment specifying what checks are done and what checks are not done.